### PR TITLE
40 mview patch update version to be merged

### PR DESCRIPTION
**Pull request by skovalenk**
### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/404
https://github.com/magento-architects/architecture/pull/3
https://github.com/magento-commerce/magento2-infrastructure/pull/1377

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by skovalenk** [comment]
@kandy 
can you take a look of this?

**Review by kandy — APPROVED** [review]

**Comment by mslabko** [comment]
@magento run all tests

File: app/code/Magento/Eav/Model/Mview/ChangeLogBatchWalker.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+namespace Magento\Eav\Model\Mview;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Sql\Expression;
+use Magento\Framework\Mview\View\ChangeLogBatchWalkerInterface;
+use Magento\Framework\Mview\View\ChangelogInterface;
+
+/**
+ * Class BatchIterator
```
Comment (mslabko) [inline app/code/Magento/Eav/Model/Mview/ChangeLogBatchWalker.php:15]: @skovalenk please add meaningful description

File: app/code/Magento/Eav/Model/Mview/ChangeLogBatchWalker.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+     *
+     * @param int $numberOfAttributes
+     * @return void
+     * @throws \Exception
+     */
+    private function setGroupConcatMax(int $numberOfAttributes): void
+    {
+        $connection = $this->resourceConnection->getConnection();
+        $connection->query(sprintf(
+            'SET SESSION %s=%s',
```
Comment (mslabko) [inline app/code/Magento/Eav/Model/Mview/ChangeLogBatchWalker.php:85]: can you rollback value after done iteration?

File: lib/internal/Magento/Framework/Mview/View/AdditionalColumnProcessorInterface.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Framework\Mview\View;
+
+use Magento\Framework\DB\Ddl\Table;
+
+interface AdditionalColumnProcessorInterface
```
Comment (mslabko) [inline lib/internal/Magento/Framework/Mview/View/AdditionalColumnProcessorInterface.php:11]: add explanation of what this interface do

File: lib/internal/Magento/Framework/Mview/etc/mview.xsd [hunk 4]
```diff
@@ -46,6 +46,7 @@
         <xs:attribute name="id" type="xs:string" use="required" />
         <xs:attribute name="class" type="classType" use="required" />
         <xs:attribute name="group" type="xs:string" use="required" />
+        <xs:attribute name="walker" type="classType" default="Magento\Framework\Mview\View\ChangeLogBatchWalker" />
```
Comment (mslabko) [inline lib/internal/Magento/Framework/Mview/etc/mview.xsd:49]: Please, add in top of the file configuration with an example of usage of new features

**Comment by lykhachov** [comment]
@magento run Static Tests

**Comment by mslabko** [comment]
@magento run all tests

**Comment by lykhachov** [comment]
@magento run Static Tests

**Comment by lykhachov** [comment]
@magento run all tests

**Comment by lykhachov** [comment]
@magento run all tests

**Comment by lykhachov** [comment]
@magento run all tests

**Comment by lykhachov** [comment]
@mgento run all tests

**Comment by lykhachov** [comment]
@magento run all tests

**Comment by mslabko** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by mslabko** [comment]
@magento run Integration Tests

**Comment by RuslanKostiv1** [comment]
@magento run Static Tests

**Comment by RuslanKostiv1** [comment]
@magento run  Functional Tests CE

**Comment by RuslanKostiv1** [comment]
@magento run Functional Tests CE, Static Tests

**Review by RuslanKostiv1 — APPROVED** [review]

**Comment by RuslanKostiv1** [comment]
@magento run Integration Tests

**Comment by mslabko** [comment]
"Semantic Version Checker - Xsd changes (MAJOR)" approved

**Comment by RuslanKostiv1** [comment]
@magento run Integration Tests

**Comment by RuslanKostiv1** [comment]
@magento import pr to magento-commerce/magento2ce

**Comment by magento-engcom-team** [comment]
@RuslanKostiv1 the pull request successfully imported.

**Comment by RuslanKostiv1** [comment]
@magento run Integration Tests

**Comment by RuslanKostiv1** [comment]
@skovalenk please merge the latest mainline and fix failed tests
keep eye on `\Magento\Framework\Mview\View\Subscription::__construct`, you need to pass `
            $ignoredUpdateColumnsBySubscription,
            $mviewConfig`
there 
```
public function __construct(
        ResourceConnection $resource,
        \Magento\Framework\DB\Ddl\TriggerFactory $triggerFactory,
        \Magento\Framework\Mview\View\CollectionInterface $viewCollection,
        \Magento\Framework\Mview\ViewInterface $view,
        $tableName,
        $columnName,
        MetadataPool $metadataPool,
        $entityInterface = null,
        $ignoredUpdateColumns = [],
        $ignoredUpdateColumnsBySubscription = [],
        Config $mviewConfig = null
    ) {
        parent::__construct(
            $resource,
            $triggerFactory,
            $viewCollection,
            $view,
            $tableName,
            $columnName,
            $ignoredUpdateColumns,
            $ignoredUpdateColumnsBySubscription,
            $mviewConfig
        );
        $this->ignoredUpdateColumns = $ignoredUpdateColumns;
        $this->entityMetadata = $metadataPool->getMetadata($entityInterface);
    }
```

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@skovalenk Unit test still failing, 

Functional tests (EE/B2B) are failing too 

```
Interceptors generation... 5/9 [=======>-----]  55% 51 secs 276.0 MiB
Area configuration aggregation... 5/9 [=======>-----]  55% 51 secs 276.0 MiB
Area configuration aggregation... 6/9 [========>----]  66% 57 secs 388.0 MiB
Interception cache generation... 6/9 [========>----]  66% 57 secs 388.0 MiBErrors during compilation:
	Magento\CatalogStaging\Model\Mview\View\Attribute\Subscription
		Incompatible argument type: Required type: array. Actual type: \Magento\Framework\Mview\Config; File:
/var/www/html/app/code/Magento/CatalogStaging/Model/Mview/View/Attribute/Subscription.php

Total Errors Count: 1

In Log.php line 92:
```
please rerun tests after fixes

**Comment by skovalenk** [comment]
Hi Ruslan I`m looking into that

**Comment by skovalenk** [comment]
@magento run all tests

**Comment by skovalenk** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@magento run all tests

**Comment by mslabko** [comment]
@magento run Unit Tests

**Comment by skovalenk** [comment]
@magento run all tests

**Comment by skovalenk** [comment]
@magento run Unit Tests

**Comment by skovalenk** [comment]
@magento run Functional Tests EE, Functional Tests CE, Functional Tests B2B

**Comment by skovalenk** [comment]
@magento run Functional Tests B2B, Functional Tests CE

**Comment by skovalenk** [comment]
@magento run all tests

**Comment by mslabko** [comment]
@magento run all tests

**Comment by mslabko** [comment]
@magento run Integration Tests

**Comment by mslabko** [comment]
@magento run all tests

**Comment by skovalenk** [comment]
@magento run all tests

**Comment by aleksejsbaranovs10** [comment]
@magento run all tests

**Comment by aleksejsbaranovs10** [comment]
@magento run Integration Tests

**Comment by RuslanKostiv1** [comment]
@magento import pr to magento-commerce/magento2ce

**Comment by magento-engcom-team** [comment]
@RuslanKostiv1, an error occurred during the Pull Request import.

**Comment by RuslanKostiv1** [comment]
already merged

**Pull request closed without merging**